---
title: "Bird Park Environmental Microbiomes"
author: "Dustin Foote, Chris Balakrishnan, Ariane Peralta"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
header-includes:
  - \usepackage{array}
  - \usepackage{graphics}
  - \usepackage[utf8]{inputenc}
output: 
  pdf_document:
  fig_caption: true
---

Project Description: Fill out

# Initial Setup
```{r Initial Setup, include=FALSE}
rm(list=ls())
setwd("~/GitHub/SylvanHeightsMyco/analyses")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code Dependencies
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
require("vegan")
require("nlme")
require("reshape")
require("ggplot2")
require("ade4")
require("tidyr")
require("reshape2")
require("picante")
```
#Import Files
## Environmental Data
```{r Import Files - Env, include=FALSE}
# Import Environmental Data
design <- read.csv("../data/SH_design.csv", header=TRUE)
```


## Microbial Data
```{r Import Files - Bacteria, include=FALSE}
# Import OTU data
# Import Raw Data
otu.in <- read.otu("../data/SH.opti_mcc.shared")

# Remove OTUs with less than two occurences across all sites
otus <- otu.in[, which(colSums(otu.in) >= 2)]

# Make Presence Absence Matrix
dataPA <- (otus > 0) * 1

# Make Relative Abundance Matrices
dataREL <- otus
for(i in 1:dim(otus)[1]){
  dataREL[i,] <- otus[i,]/sum(otus[i,])
}

# Import Taxonomy File
otu.tax <- read.tax(taxonomy = "../data/SH.opti_mcc.0.03.cons.taxonomy",
                   format = "rdp", tax.levels = 6, col.tax = 3)
```
# Diversity Metrics - Hypothesis Testing
```{r Diversity Metrics - Bacteria, echo=TRUE}
# Rarefy Abundances (min abundance is 8106. We are sampling to 8000)
min(rowSums(otus))
max(rowSums(otus))
mean(rowSums(otus))
SH.r <- rrarefy(otus, 8000)

# Fisher's Alpha
fisher <- fisher.alpha(SH.r)

# Species Richness
richness <- rowSums((SH.r >= 1))

# Shannon Diversity
shannon <- diversity(SH.r, "shannon")

# Simpson's Evenness
simp.even <- apply(SH.r, 1, simp_even)

#Pielouâ€™s evenness
J <- shannon/log(specnumber(SH.r[,-c(1:1)]))

#combined richness, diversity, evenness
diversity <- cbind(design,richness,shannon,simp.even,J)
```

# Simple Hypothesis Testing - Microbes
```{r perMANOVA - Bacteria - , echo=TRUE}
#PERMANOVA 
new.data <-cbind(design,dataREL)
adonis = adonis(new.data[,-c(1:6)]~Transect*Location, method = "bray", data = new.data, perm=1000)
adonis
```
# Microbial Ordinations

## Principal Coordinates Ordination
```{r Ordination (PCoA) - Bacteria - all years, include=FALSE}
# Principal Coordinates Analysis
dataREL.dist <- vegdist(dataREL, method="bray")

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #30.2
explainvar2b #21.6
```

```{r Ordination (PCoA) - Bacteria -, include=FALSE}
pcoa.groups <- paste(new.data$Location, new.data$Transect, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # Transect
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Location

# Principal Coordinates Analysis 
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_point(aes(fill=pcoa.col), colour = "black", size=5, stroke = 0.75) + 
  scale_colour_manual(labels = c("SA","SOURCE", "WWWD"), 
                    values = c("#33CCFF", "#CCCCCC","#336600")) + 
  scale_fill_manual(labels = c("SA","SOURCE", "WWWD"), 
                    values = c("#33CCFF", "#CCCCCC","#336600"))  + 
  scale_shape_manual(labels = c("bird","exit","pre","source"),
                    values = c(21, 24, 22, 25)) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") +
  theme(axis.title = element_text(size=18), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (30.2%)") + ylab("PCoA 2 (21.6%)") + 
  labs(fill = "Location", shape = "Transect") +
  guides(fill = guide_legend(override.aes = list(pch=21, size = 4, colour="black")))
         #shape = guide_legend(override.aes = list(size = 4, fill="black")))
ggsave("../figures/16SrRNA_SH.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)

```

# OTU graph
```{r Myco graph, include=FALSE}
#NOTE: 
#Otu0113 (451 reads)
#Otu1461 (12 reads)
#Otu2496 (6 reads)
#Otu3114 (4 reads)

p <- ggplot(new.data, aes(x=Transect, y=Otu0113, shape=as.factor(Location), color=as.factor(Transect), fill=as.factor(Transect))) + stat_summary(fun.data=mean_cl_boot,size=0.75) + scale_shape_manual(name="Location", values=c(21, 24, 22, 23)) +scale_color_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD")) +scale_fill_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD"))   

p + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.75, 
          hjust=0.6, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Transect", y = "Rel. Abun. Mycobacteria OTU0113") 
  
ggsave("../figures/OTU0113.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(new.data, aes(x=Transect, y=Otu1461, shape=as.factor(Location), color=as.factor(Transect), fill=as.factor(Transect))) + stat_summary(fun.data=mean_cl_boot,size=0.75) + scale_shape_manual(name="Location", values=c(21, 24, 22, 23)) +scale_color_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD")) +scale_fill_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD"))   

p + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.75, 
          hjust=0.6, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Transect", y = "Rel. Abun. Mycobacteria OTU1461") 
  
ggsave("../figures/OTU1461.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(new.data, aes(x=Transect, y=Otu2496, shape=as.factor(Location), color=as.factor(Transect), fill=as.factor(Transect))) + stat_summary(fun.data=mean_cl_boot,size=0.75) + scale_shape_manual(name="Location", values=c(21, 24, 22, 23)) +scale_color_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD")) +scale_fill_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD"))   

p + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.75, 
          hjust=0.6, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Transect", y = "Rel. Abun. Mycobacteria OTU0113") 
  
ggsave("../figures/OTU2496.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)

p <- ggplot(new.data, aes(x=Transect, y=Otu3114, shape=as.factor(Location), color=as.factor(Transect), fill=as.factor(Transect))) + stat_summary(fun.data=mean_cl_boot,size=0.75) + scale_shape_manual(name="Location", values=c(21, 24, 22, 23)) +scale_color_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD")) +scale_fill_manual(name="Transect", values=c("#33CCFF", "#CCCCCC","#336600"), labels = c("SA","SOURCE","WWWD"))   

p + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.75, 
          hjust=0.6, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Transect", y = "Rel. Abun. Mycobacteria OTU0113") 
  
ggsave("../figures/OTU3114.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=NA, height=NA, dpi=300, limitsize=TRUE)
```

# Bacterial community indicator species analysis 2013 only - plant
```{r Bacteria Indicator Species, include=FALSE}
new.data <-cbind(design,dataREL)
library("labdsv")

design.type <- new.data$Transect

dataREL.ind <- dataREL[, colSums(dataREL) > 0.05]
bac.ind <- indval(dataREL.ind, design.type)
levels(design.type)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- otu.tax[which(as.character(otu.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_Location.txt",
            sep="\t", row.names = F, quote = F)
```
